app/services/SubmissionService.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import com.google.inject.Inject
+import config.AppConfig
+import connectors.CustomsDeclareExportsConnector
+import javax.inject.Singleton
+import metrics.ExportsMetrics
+import metrics.MetricIdentifiers.submissionMetric
+import models.requests.JourneyRequest
+import uk.gov.hmrc.http.cache.client.CacheMap
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import controllers.util.CacheIdGenerator.cacheId
+import play.api.Logger
+import play.api.http.Status.ACCEPTED
+import services.audit.{AuditService, AuditTypes}
+import services.audit.EventData._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SubmissionService @Inject()(
+  appConfig: AppConfig,
+  cacheService: CustomsCacheService,
+  exportsConnector: CustomsDeclareExportsConnector,
+  auditService: AuditService,
+  exportsMetrics: ExportsMetrics
+) {
+
+  def submit(
+    cacheMap: CacheMap
+  )(implicit request: JourneyRequest[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] = {
```

Comment:
Service shouldn't have any request, headerCarrier, executionContext. All of this should be handled in the controller.

Reply:
I agree with this approach. We should not leak http request middleware to the business logic service.
Advantages: 
 - Easier to maintain/read
 - Easier to unit test